File: src/main/org/nlogo/prim/_constdouble.java
// (C) 2012 Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim;

import org.nlogo.api.Syntax;
import org.nlogo.nvm.Context;
import org.nlogo.nvm.Pure;
import org.nlogo.nvm.Reporter;

public final strictfp class _constdouble
    extends Reporter implements Pure {
  final Double value;
  public final double primitiveValue;

  public _constdouble(Double value) {
    this.value = value;
    primitiveValue = value.doubleValue();
  }

  @Override
  public Syntax syntax() {
    return Syntax.reporterSyntax(Syntax.NumberType());
  }

  @Override
  public String toString() {
    return super.toString() + ":" + primitiveValue;
  }

  @Override
  public Object report(Context context) {
    return value;
  }

  public Double report_1(Context context) {
    return value;
  }

  public double report_2(Context context) {
    return primitiveValue;
  }
}

File: src/main/prim/_constdouble.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.prim

import org.nlogo.api.Syntax
import org.nlogo.nvm.{ Reporter, Pure, Context }

class _constdouble(value: java.lang.Double) extends Reporter with Pure {

  private[this] val _primitiveValue = value.doubleValue
  def primitiveValue = _primitiveValue

  override def syntax =
    Syntax.reporterSyntax(Syntax.NumberType)

  override def toString =
    super.toString + ":" + primitiveValue

  override def report(context: Context): java.lang.Double =
    report_1(context)

  def report_1(context: Context): java.lang.Double =
    value

  def report_2(context: Context): Double =
    _primitiveValue

}
